### PR TITLE
Bugfix/kill copy on disconnect

### PR DESCRIPTION
**Pull request by kuenishi**
Copying can't be stopped on client disconnect, because it does not need any socket read/writes once copying started. That makes detecting client disconnect harder - this patch puts a zero-timeout read on socket to find `{error, closed}` and other errors. This is a dirty hack and mochiweb or webmachine should have such interface that handles lower layer connection error handling, but for now we adopt this hack, which should be fixed in the future. 
Copy should be cancelled because orphan copy without clients are likely not required by clients and may cause overloaded cluster.

Important limitation is: if client sends any additional HTTP request to CS during copy being processed, Riak CS reads that request and throws error, and cancels the copy.

### CODE REVIEW

File: src/riak_cs_copy_object.erl [hunk 1]
```diff
@@ -189,15 +196,24 @@ get_content_md5({_MD5, _Str}) ->
...
-            riak_cs_get_fsm:stop(GetPid),
-            riak_cs_put_fsm:finalize(PutPid, MD5);
-        {chunk, Block} ->
-            riak_cs_put_fsm:augment_data(PutPid, Block),
-            get_and_put(GetPid, PutPid, MD5)
+-spec get_and_put(pid(), pid(), list(), fun(() -> boolean()))
+                 -> {ok, lfs_manifest()} | {error, term()}.
+get_and_put(GetPid, PutPid, MD5, ContFun) ->
+    case ContFun() of
+        true ->
```
Comment (shino) [inline src/riak_cs_copy_object.erl]: A trailing white space here.

**Comment by kuenishi** [comment]
Add note: detection only with `inet:peername/1` didn't work for me, with [this server](https://gist.github.com/kuenishi/c1363dd5021c29a0292a) , with my linux. That is why using `gen_tcp:recv/2` for disconnect detection. Maybe peername is cached inside the socket?

**Comment by kuenishi** [comment]
Fixed :)

**Comment by shino** [comment]
@borshop merge

**Pull request merged**